### ISSUE TITLE
[BUG] Lineage splash screen on Wayland session start takes up the whole screen

### ISSUE THREAD

**Issue by justin13888** [waydroid/waydroid#1182]
### Describe the bug

I like to use Waydroid in multi-window mode (i.e. with apps opening independently in different windows rather than the full screen UI).

I've changed a property as instructed here: <https://docs.waydro.id/usage/install-on-desktops#launch-waydroid-in-multi-window-mode>

The "bug"  that I'm having is that whenever Waydroid starts the session for the first time, it covers my screen for 5-10 seconds with the lineage boot animation. I tried googling but cannot find documentation addressing that so I've submitted a bug report here. Appreciate any help!

### Waydroid version

1.4.2

### Device

Linux Desktop

### Operating System

Fedora 39 (Workstation Edition)

### Kernel version

6.5.12-tkg-eevdf (custom kernel)

### Desktop Environment

GNOME 45

### GPU

AMD ATI Radeon RX 7900 XT

### Logs

/var/lib/waydroid/waydroid.cfg
```
[waydroid]
arch = x86_64
vendor_type = MAINLINE
system_datetime = 1700271104
vendor_datetime = 1700306537
suspend_action = freeze
mount_overlays = True
images_path = /var/lib/waydroid/images
system_ota = https://ota.waydro.id/system/lineage/waydroid_x86_64/GAPPS.json
vendor_ota = https://ota.waydro.id/vendor/waydroid_x86_64/MAINLINE.json
binder = anbox-binder
vndbinder = anbox-vndbinder
hwbinder = anbox-hwbinder
binder_protocol = aidl3
service_manager_protocol = aidl3

[properties]

```

/var/lib/waydroid/waydroid.prop

```
sys.use_memfd=true
debug.stagefright.ccodec=0
ro.hardware.gralloc=gbm
ro.hardware.egl=mesa
ro.hardware.vulkan=radeon
ro.hardware.camera=v4l2
ro.opengles.version=196609
waydroid.system_ota=https://ota.waydro.id/system/lineage/waydroid_x86_64/GAPPS.json
waydroid.vendor_ota=https://ota.waydro.id/vendor/waydroid_x86_64/MAINLINE.json
waydroid.tools_version=1.4.2
ro.vndk.lite=true
waydroid.host.user=justin13888
waydroid.host.uid=1000
waydroid.host.gid=1000
waydroid.host_data_path=/home/justin13888/.local/share/waydroid/data
waydroid.xdg_runtime_dir=/run/user/1000
waydroid.pulse_runtime_path=/run/user/1000/pulse
waydroid.wayland_display=wayland-0
waydroid.background_start=false
waydroid.stub_sensors_hal=1
```

/var/lib/waydroid/waydroid_base.prop

```
sys.use_memfd=true
debug.stagefright.ccodec=0
ro.hardware.gralloc=gbm
ro.hardware.egl=mesa
ro.hardware.vulkan=radeon
ro.hardware.camera=v4l2
ro.opengles.version=196609
waydroid.system_ota=https://ota.waydro.id/system/lineage/waydroid_x86_64/GAPPS.json
waydroid.vendor_ota=https://ota.waydro.id/vendor/waydroid_x86_64/MAINLINE.json
waydroid.tools_version=1.4.2
ro.vndk.lite=true
```

I've excluded the following log data because I don't think the logs are really long and does not appear to be relevant...

/var/lib/waydroid/waydroid.log
`sudo waydroid shell -- logcat -d | tee logcat.txt`
`sudo dmesg | tee dmesg.txt`

**Comment by aleasto** [comment]
You can make waydroid start in the background instead of in the forground with `waydroid session start`

**Comment by justin13888** [comment]
That technically works. I hope others who have a similar issue either finds this workaround or I'd like to suggest a better way to approach this.

@aleasto Your suggestion works but requires the waydroid session to be spun up even when you aren't using an app on waydroid. While I personally don't care because my system is not resource constrained, it seems more appropriate to have a config to make the splash screen either be hidden or for an option like a smaller splash animation (that doesn't take up the whole screen).

**Comment by superboo07** [comment]
please add the option to restore it being hidden

**Comment by sidevesh** [comment]
+1, please make it configurable, its okay to by default show the splash screen but allow it to be disabled,
especially when multi_windows is enabled its really jarring to get a whole screen black booting animation suddenly.

**Comment by sidevesh** [comment]
@aleasto can you point to the commit were this was changed from the older behavior to now showing the splash screen ?
I can take a look at making it a configurable parameter in a PR if y'all agree with that config parameter to be added.

**Comment by sidevesh** [comment]
found it, https://github.com/waydroid/waydroid/commit/204403056e6c8adc14406be5e64c4f1996e185bc#diff-fe3a1a33a19f8ca8be67e79be901ebe3922c786a07d176ec65c9f83f17f67537R69
changing the line `tools.actions.session_manager.start(args, launchNow, background=False)` to `tools.actions.session_manager.start(args, launchNow, background=True)` in `tools/actions/app_manager.py` fixed it for me,
@aleasto is `background_start_if_session_not_running` as a config parameter okay ? I can do  a PR for this if that works or lmk if you would rather handle this in some other way.

**Comment by sidevesh** [comment]
or we can make it so background=True is passed if multi_windows is enabled ? splash screen should be less jarring if multi_windows is not being used

**Comment by RokeJulianLockhart** [comment]
I don't believe that hiding the window is the solution. It should merely render in a real window of standard size. That provides the progress feedback, and doesn't cause the entire display to be unusable. I know that as a "solution", I wouldn't want it to be hidden.

**Comment by sidevesh** [comment]
I don't think it would be possible to show the splash screen in a small window if I understand correctly how waydroid works,
I agree that some form of feedback while waydroid session starts would help but a full screen splash screen is not good for someone just trying to launch an app,
its jarring and takes away from the seamlessness of waydroid especially  when used with `multi_windows` enabled.
While some form of minimal launch feedback is figured out I think having a configurable option to hide the full screen splash screen would be great, even if its opt-in only.

**Comment by beedell-roke** [comment]
> I don't think it would be possible to show the splash screen in a small window if I understand correctly how waydroid works.

@sidevesh, do you mean that AOSP doesn't support dynamic monitor resize (on its primary display)?

- It does on external display (re/dis)-connect, at least. I know because I've used external monitors, and it has to handle them disconnecting at will.

- Resizing the resolution of the display during runtime should be possible with `adb`, per [this Reddit comment](https://www.reddit.com/r/pokemongodev/s/5yaF58sSij), if that's what we want.

**Comment by erbold-bu** [comment]
No developments here?

**Comment by beedell-roke** [comment]
@erbold-bu, you see none here, so no.
